firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/crcelements.txt

```diff
@@ -0,0 +1,131 @@
+/* compute_body_crc
+Scenario: CRCElement: The compute_body_crc method correctly computes the CRC of a payload from a CRCElement body given as input
+  GIVEN("A CRC element initialized with an empty payload of capacity 250 bytes")
+    WHEN("the crc is computed on an input_buffer with empty payload")
+      THEN("the computed CRC is '0x00000000'")
+
+    WHEN("the crc is computed on an input_buffer with payload '123456789'")
+      THEN("the computed crc is '0xe3069283'")
+
+    WHEN("the CRC is computed on an input_buffer with payload '0x00' ")
+      THEN("the computed crc is '0x527d5351'")
+
+    WHEN("the crc is computed on an input_buffer with payload '0x01' ")
+      THEN("the computed crc is '0xa016d052'")
+
+/* write function
+Scenario: ConstructedCRCElement: The write method correctly generates a body from the payload given in the constructor
+  GIVEN("A crc element initalised with payload=[0x01, 0x02, 0x05]")
+    WHEN("The crc and payload are written to the output buffer")
+      THEN("the crc accessor method returns 0 before the write method is called")
+      THEN("After the write method is called, the crc accessor method returns a value equal to the result of directly computing a CRC on the payload.")
+      THEN("The CRC field of the body's header matches the value returned by the crc accessor method.")
+      THEN("The body's payload section correctly stores the payload as 0x01 0x02 0x05")
+      THEN("The output buffer is as expected")
+
+  GIVEN("A crc element is initalised with an empty payload")
+    WHEN("The crc and payload are written to the output buffer")
+      THEN("the crc accessor method returns 0 before the write method is called")
+      THEN("The CRC field of the body's header matches the value returned by the crc accessor method.")
+      THEN("The body's payload section is empty")
+      THEN("The output buffer is as expected")
+
+/* parse function
+Scenario: CRCElement: The parse method correctly updates its internal crc value and the constructor's payload buffer from the input buffer
+  GIVEN("A CRCElement constructed with an empty payload buffer with a capacity of 254 bytes")
+
+    WHEN("A body without a complete 4-byte header is parsed")
+      THEN("the parse method reports out of bounds status")
```

Comment:
I think we should also add a `THEN("the constructor's payload buffer remains unchanged")`

Reply:
Added.

Reply:
It looks like this change got reverted somehow?